practice/031/answer/impl.ml

```diff
@@ -0,0 +1,13 @@
+let is_prime n =
+  let n = abs n in
+  let rec is_not_divisor d =
+    d * d > n || (n mod d <> 0 && is_not_divisor (d + 1)) in
+  n <> 1 && is_not_divisor 2
+
+let rec flag_prime n lst_n = 
+  match lst_n with 
+  | [] -> []
+  | hd :: tl -> (hd, is_prime hd) :: flag_prime n tl
+
+
+  let prime_0_to_100 = [2; 3; 5; 7; 11; 13; 17; 19; 23; 29; 31; 37; 41; 43; 47; 53; 59; 61; 67; 71; 73; 79; 83; 89; 97]
```

Comment:
```suggestion
  let prime_0_to_100 = [2; 3; 5; 7; 11; 13; 17; 19; 23; 29; 31; 37; 41; 43; 47; 53; 59; 61; 67; 71; 73; 79; 83; 89; 97]

```

Reply:
Function `flag_prime` and list `prime_0_to_100` (that should be renamed `primes`) should be defined in file `ex.ml`, not in `impl.ml` they should be used to test `is_prime`. Also, because of `flag_prime` should be used to test `is_prime`, it must not use it.

`flag_prime` is basically iterating from 0 to the latest value in `primes`. Each value is compared to the head of the primes list
- if equal then, the result pair is marked `true` and the head of `primes` is removed
- if no equal, the result pair is marked `false` an computation continues with `primes` unchanged